src/modules/storage/BlockStorage.ts

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+
+    The class that insert and read the block into the database.
+
+    Now only block header are stored.
+    Transactions and other data will be managed by different classes.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import * as sqlite from 'sqlite3';
+
+/**
+ * The class that insert and read the block into the database.
+ */
+class BlockStorage
+{
+    /**
+     *  The instance of sqlite3
+     */
+    private db: sqlite.Database;
+
+    /**
+     * Constructor
+     * @param filename Valid values are filenames,
+     * ":memory:" for an anonymous in-memory database and
+     * an empty string for an anonymous disk-based database
+     * @param callback : If provided, this function will be called when
+     * the database was opened successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    constructor (filename : string, callback : any = null)
+    {
+        this.db = new sqlite.Database(filename,
+            sqlite.OPEN_CREATE | sqlite.OPEN_READWRITE |
+            sqlite.OPEN_SHAREDCACHE);
+        this.db.configure("busyTimeout", 1000);
+        this.createTable(callback);
+    }
+
+    /**
+     * Creates a table. table name is `block`.
+     * @param callback If provided, this function will be called when
+     * the database was finished successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    public createTable (callback : any = null)
```

Comment:
```suggestion
    public createTables (callback : any = null)
```
Or setTables Or initialTables

I think we will define the table schema here.

Reply:
When the constructor runs, it creates a table if it does not exist. Therefore, when an API server is installed, there is no need to do additional work on table creation.

Reply:
Then the method shouldn't be `public`. It shouldn't be used inside the class. And it is not generic, so the name should reflect that. E.g. `createBlocksTable`